### ISSUE TITLE
Upgrade Material Design icons to latest version

### ISSUE THREAD

**Issue by cryocaustik** [squidfunk/mkdocs-material#1174]
## Description

When using the option of `icon` under the theme, certain material icons will either not show at all, or will show with incorrect padding.

### Expected behavior

Nav Icon to be replaced with another Material Icon, using the same padding as the default icon.

![image](https://user-images.githubusercontent.com/15346996/61245305-0b6cdb00-a701-11e9-888d-ef9108d8ee31.png)

### Actual behavior

Will either not show the icon, incorrectly pad the icon, or both.

![image](https://user-images.githubusercontent.com/15346996/61245333-1b84ba80-a701-11e9-99aa-6594092d2b3a.png)
![image](https://user-images.githubusercontent.com/15346996/61245342-1fb0d800-a701-11e9-9cd8-7c0a179b72b8.png)

### Steps to reproduce the bug

1. adjust `mkdocs.yml` to use

```yml
theme:
  name: material
  logo:
    icon: bar_chart
```

2. run `mkdocs serve`
3. preview site in browser

### Package versions

* Python: 3.7.3
* MkDocs: 1.0.4
* Material: 4.4.0

### Project configuration

``` yaml
The contents of your mkdocs.yml
```

### System information

* OS: Windows 10 Pro x64
* Browser: Chrome, Firefox Developer Edition

**Comment by squidfunk** [comment]
I would guess that the icons you're trying to use are part of a newer version of the Material icon set. As the icon set is [bundled with this theme](https://github.com/squidfunk/mkdocs-material/tree/master/src/assets/fonts) (for reasons of compliance with GDPR, see #739), so we most likely need to update the bundled icons. As I'm low on time, I'm happy to merge PRs.

The padding originates from the fact that you use the glyph integration (using `insert_chart_outlined` instead of raw code points), so if the glyph is not found (or only a partial match), the rest of the characters of the glyph are printed as missing code points, resulting in the extra whitespace. This is the reason why the Material theme uses code points, as they're always one "character" wide, e.g.:

https://github.com/squidfunk/mkdocs-material/blob/47856bc323eb9b3c6b1330f4441e94524559e7eb/src/partials/search.html#L38

**Comment by squidfunk** [comment]
Updating the Material Icon font to the latest version is not as easy as I thought. The problem is that the last version which Google released for self-hosting is [2.2.0](https://github.com/google/material-design-icons/tree/master/iconfont). [This StackOverflow comment](https://graphicdesign.stackexchange.com/a/132449) sheds some more light on the problem. Since we're including the Material icons with the theme for GDPR reasons (see #739), we're pretty much stuck with the current version.

We could – in theory – just download the fonts from the URLs Google uses for hosting the specimen (as listed in the StackOverflow comment above), but I don't know if this might violate their terms of use. There could be a reason why Google doesn't upload the self-hosted version anymore.

I don't really know how to proceed here except leaving the implementation as-is.

**Comment by squidfunk** [comment]
As a further note: the original problem described stems from the fact that the _ligature_ `insert_chart_outlined` is not part of the Material font distribution we're using, which is the official one. If you experience a weird padding next to the logo, it's related to a part of the _ligature_ not being detected/mapped or not being present at all.

While `insert_chart` works, `insert_chart_outlined` will translate to `[ICON GLYPH]_outlined`, resulting in the additional padding induced by `_outlined`.

This is not a bug in this theme but a configuration error (since an unknown ligature is used). As we cannot upgrade the icon font (see my comment above), there's nothing we can do at this point. I'm sorry. Maybe Google will release a new version of the font in the future, so we can upgrade it. Until then, I will close this issue.

**Comment by squidfunk** [comment]
I have good news! As the original repository hasn't been updated, I searched for another way of grabbing the latest icons from [material.io](https://material.io/resources/icons/?style=baseline) and found an npm package that seems to download the SVG versions automatically from their site. While the web font seems to contain around 400 icons, the SVG icon set contains a whopping 3.8k icons! I started inlining the icons as SVGs and it works like a charm. They will all be bundled with the next version of Material. Including the FontAwesome icon set the theme will contain 5.3k icons!

Regarding the license - material.io states that the icons are all available under Apache license which is compatible with MIT which Material is using, so there should be no problem.
